svc.go

```diff
@@ -123,7 +126,18 @@ func (as *autoNATService) handleDial(p peer.ID, obsaddr ma.Multiaddr, mpi *pb.Me
 		}
 
 		if as.config.dialPolicy.skipDial(addr) {
-			continue
+			if manet.IsPrivateAddr(addr) {
+				addrcodename, addrport, err := as.extractPort(addr)
+				if err == nil && addrcodename == obscodename && addrport != obsport { //make a new address
+					obsportstr := fmt.Sprintf("/%s/%s", obscodename, obsport)
+					addrportstr := fmt.Sprintf("/%s/%s", addrcodename, addrport)
+					addr, err = ma.NewMultiaddr(strings.Replace(obsaddr.String(), obsportstr, addrportstr, -1)) //replace the private addr
```

Comment:
this could have issues with circuit addresses or other weird edges of multiaddr. rather than string replace prefer finding the component with the expected code, and changing it in that way using the multiaddr library for parsing.

Reply:
The nat ```obsaddr`` shouldn't be a circuit address or over the proxy, and it supposed to be a simple /ipv4 or /ipv6 address in this situation.

The ```go-multiaddr``` package didn't provide a function to replace a component and I have to rewrite all bytes and parse to multiaddr, so replace the ```/tcp/port``` could be a clean way to resolve the issue, IMO.